api/common/ctx.go

```diff
@@ -9,6 +9,16 @@ import (
 
 type contextKey string
 
+//RidKey is the context key to use to store a request ID
+type RidKey string
+
+const ridCtxKey = RidKey("fn_request_id")
+
+//RIDContextKey returns the context key to use for dealing with request IDs
+func RIDContextKey() RidKey {
+	return ridCtxKey
+}
```

Comment:
couple little things to clean up here. casing should be consistent on `RID`, `s/RidKey/RIDKEY/`. from usage, i'm not sure we want to provide alternate keys here? if we do, maybe it should be a configurable var (plumbing is never fun) -- this seems like a thing we should punt on since we don't need it right now.

the key is needed only in `common`, we shouldn't expose the type for that (the type is generally there to keep context value keys from conflicting if they end up being the same string/int value. e.g. `keyAType("a") != `otherKeyAType("a")`) - the methods hide the usage anyway, and shouldn't be needed. if we use a const for the `RIDContextKey` value it can be of either type we need it for in various places, as opposed to having to cast it to a string from the `agent` package (`agent` should get a `string` and not have to deal with this cast).

```go
// unexported type
type ridKey string

// since this is a const, it can be of type ridKey or string, when needed.
const RIDContextKey = "fn_request_id"
```

Reply:
it's generally the case that type declarations should be the fully named type and only local variables should have the shorthand for it, such as `rid` representing a `RequestID` - it would be great to expand the exported `RIDContextKey` to `RequestIDContextKey` or `RequestIDKey` such here, as well (for example, in the `agent` caller, it's hard to know what an `RID` is immediately upon looking at it - this should be the case.

Reply:
ok thanks. I'll use the const and get rid of the exported type. I am going to add a new method in common called WithRequestID to store a request ID into the context.